test/e2e/helpers.go

```diff
@@ -36,14 +36,13 @@ type K8sDeleteFunc func(ctx context.Context, name string, options metav1.DeleteO
 // asserting there were no errors.
 func GetK8sObjectWithRetry[T kruntime.Object](
 	ctx context.Context, getFunc K8sGetFunc[T], name string, options metav1.GetOptions,
-) T {
-	var object T
+) (result T) {
```

Comment:
Both. If I didn't name it it's have it I would have to add a definition for `var result T` anyway so it seemed neater to define it there.

Reply:
Well:
- we need `result` to have a broader scope than `err` so it can be returned at the end.
- we bind `err` and `result` at the same time on line 42. 

AFAIK because of the above if we use `:=` we redefine the scope of `result` and then the final return can't work. This is my understanding of the compiler errors I see when I tried to implement this but I might have this wrong.